ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java

```diff
@@ -78,6 +78,16 @@
 
 public class XARecoveryModule implements ExtendedRecoveryModule
 {
+	private static AtomicBoolean recoveryProblems = new AtomicBoolean(false);
```

Comment:
why this needs to be `static`? I'm asking this question as I'm interested if there is some reasons behind this decision.
I assume the approach should be to not use the `static` when it's not particularly needed. The API which the behaviour exposing to outer world - aka API that was used is the call to `XARecoveryModule.isPeriodcWorkSuccesful`.
This one is then consumed by `WorkerService` (https://github.com/jbosstm/narayana/blob/5.10.4.Final/ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/WorkerService.java#L114) to inform about issue on recovery process when verbose output is demanded.

Reply:
Oh! sorry, I missed to see that `XARecoveryModule.isPeriodcWorkSuccesful` is already available, in order to expose the recovery state. I will remove static keyword. Thanks